src/cli/commands/import.js

```diff
@@ -150,28 +153,51 @@ class ImportResolver extends BaseResolver {
     return this.resolveImport(info);
   }
 
+  async resolveFixedVersion(fixedVersionPattern: string): Promise<?Manifest> {
+    const {range} = normalizePattern(fixedVersionPattern);
+    const exoticResolver = getExoticResolver(range);
+    const manifest = exoticResolver
+      ? await this.request.findExoticVersionInfo(exoticResolver, range)
+      : await this.request.findVersionOnRegistry(fixedVersionPattern);
+    return manifest;
+  }
+
   async resolve(): Promise<Manifest> {
-    const {name} = normalizePattern(this.pattern);
-    let cwd = this.getCwd();
-    while (!path.relative(this.config.cwd, cwd).startsWith('..')) {
-      const loc = path.join(cwd, 'node_modules', name);
-      const info = await this.config.getCache(`import-resolver-${loc}`, () => this.resolveLocation(loc));
+    if (this.request instanceof ImportPackageRequest && this.request.fixedVersionPattern) {
```

Comment:
Could we refactor these cases into separate functions for readability/clarity? something like

```
async resolve(): Promise<Manifest> {
  if (this.request instanceof ImportPackageRequest && this.request.fixedVersionPattern) {
    return await this._resolveFromFixedVersions();
  } else {
    return await this._resolveFromNodeModules();
  }
}
```

Reply:
Yes, I like it!